Check for null before checking for type IEnumerable.

josteink opened the pull request:
This closes https://github.com/nunit/nunit/issues/2486.

CharliePoole commented:
Looks good to me.

josteink commented:
I like that. I’ll make it so.

jnm2 commented:
:+1: `ConstraintUtils` was just off the top of my head, so if you find an existing helper class that seems related, we should probably use that.

josteink commented:
I've updated the code in the places I've found which needed changing, and I've corrected the issues identified by the CI-build.

If you think there are more changes which needs to be made, please let me know. Otherwise I'll assume this is ready for a new review :)

mikkelbu commented on src/NUnitFramework/framework/Internal/ConstraintUtil.cs:
```diff
@@ -0,0 +1,50 @@
+        /// <param name="paramName">Name of the parameter as passed into the checking method.</param>
+        /// <typeparam name="T">The type to require</typeparam>  
+        /// <returns>A properly typed object, or throws. Never null.</returns>
+        /// </summary>
```
I usually put `param`, `typeparam`, and `returns` outside the `summary` element (I don't know if this makes a difference or not). Also it is a bit inconsisten whether the the sentence ends with a period.

josteink replied:
I agree that makes sense. Will fix.

josteink commented on src/NUnitFramework/framework/Internal/ConstraintUtil.cs:
```diff
@@ -0,0 +1,50 @@
+// ***********************************************************************
+// Copyright (c) 2015 Charlie Poole, Rob Prouse
```
Fair enough. Will fix.

josteink commented:
> Also, I really like the new method. Is there any reason why you did not use it in

Yes. I realized such a reusability concern could possibly arise in the future, but decided that this could be a good "base-commit" to land properly before making any changes other aspects of the framework.

There were also the case of poor tooling. I was fixing up the PR on my home machine running whatever version of mono ships with Fedora, and I couldn't get the whole build-chain running (I multiple causes of build-failure with the Linux-solution).

I therefore decided to be defensive about it, and not touch any more components than I felt I could guarantee the correctness of.

josteink commented:
I've pushed a final commit which should address most of the concerns @mikkelbu had.

ChrisMaddock commented:
![image](https://user-images.githubusercontent.com/4837132/31612110-8a10efb8-b276-11e7-91b6-01e135a68ad1.png)

@mikkelbu - do you not see "Dismiss review" at the bottom here? We should have the same access rights, I believe.

Merging - thanks @josteink! We'd be interested to learn about/fix your mono problems - if you'd fancy debugging and opening an issue? You can see the mono versions we currently test in CI in the .travis.yml file.

ChrisMaddock commented:
(Also, @mikkelbu - can I leave it with you to open an issue re: the additional changes? Thanks!)

jnm2 commented:
Ooh, race condition! I was going to request a change but we can do it later, it's internal and minor.

jnm2 commented:
(`static` on the new class)

ChrisMaddock commented:
Ah, sorry! I was happy it was internal, so didn't read further down the line! 🙈

jnm2 commented:
@ChrisMaddock That's not actually a bad strategy! :D

mikkelbu commented:
@ChrisMaddock  I don't think I've the same rights. I went through all the open PRs and all I got was the following
![image](https://user-images.githubusercontent.com/1007631/31615158-002f81fc-b289-11e7-824d-c2681d9fc25b.png)
But accordingly to the guide (https://help.github.com/articles/dismissing-a-pull-request-review/) "repository administrators or **people with write access** can dismiss a review." I should have enough?

Ps. I'll create the new issue

ChrisMaddock commented:
@rprouse - could you investigate? Or @CharliePoole - if you're still an admin on the repo?

CharliePoole commented:
Too late to investigate this one, as it is merged! In general, I can see "Dismiss review" on both my own reviews and those of others. However, Rob and I are owners of the organization so that doesn't prove much.

ChrisMaddock commented:
Sorry - I meant could you investigate for any differences between mine and @mikkelbu's admin rights. AFAIK, they should be the same!

ChrisMaddock commented:
To be clear, currently: I can dismiss reviews and @mikkelbu can't. I believe we should have the same permissions for this!

CharliePoole commented:
I can't see any difference in GitHub. Two thoughts:

* @mikkelbu are you sure you were logged in?
* Could it be a browser issue? I use Firefox.

mikkelbu commented:
@CharliePoole I'm quite sure that I'm logged in. Also the box about "Checks and Change requests" in the bottom is not visible if one is not logged in (as far as I can tell from testing now). I've tried Chrome, IE, and Edge on all the open PRs in this project I see the same "See review", but no "Dismiss review".

CharliePoole commented:
As far as I can see, the only difference is that @ChrisMaddock is on the Core Team. But that doesn't grant any special rights as an admin or maintainer. @rprouse can you see anything?
